src/server/worker/datum/datum.go

```diff
@@ -258,6 +258,9 @@ func (d *Datum) downloadData(downloader pfssync.Downloader) error {
 				return nil
 			}),
 		}
+		if input.S3 {
+			opts = append(opts, pfssync.WithS3())
```

Comment:
Is the functional option necessary? It seems like you could just continue here?

Reply:
I just didn't quite understand the control flow here; I've removed the functional option.